viewer/packages/360-images/src/Image360Facade.ts

```diff
@@ -115,19 +115,29 @@ export class Image360Facade<T> {
   }
 
   public intersect(coords: THREE.Vector2, camera: THREE.Camera): Image360Entity | undefined {
-    this._rayCaster.setFromCamera(coords, camera);
     const cameraDirection = camera.getWorldDirection(new THREE.Vector3());
     const cameraPosition = camera.position.clone();
-
-    const intersections = this._image360Collections
-      .flatMap(getImage360Entities)
-      .filter(hasVisibleIcon)
-      .map(entity => getIntersection(entity, this._rayCaster.ray))
-      .filter(hasIntersection)
-      .map(intersectionToCameraSpace)
-      .filter(isInFrontOfCamera)
-      .sort(byDistanceToCamera)
-      .map(selectEntity);
+    const collectionMatrix = new THREE.Matrix4();
+
+    const intersections = this._image360Collections.flatMap(collection =>
+      getImage360Entities(collection)
+        .filter(hasVisibleIcon)
+        .map(
+          getIntersector(
+            getTransformedRay(
+              this._rayCaster,
+              coords,
+              camera,
+              getWorldToModelCollectionMatrix(collection, collectionMatrix)
+            )
+          )
+        )
+        .filter(hasIntersection)
+        .map(intersectionToCameraSpace)
+        .filter(isInFrontOfCamera)
+        .sort(byDistanceToCamera)
+        .map(selectEntity)
+    );
 
     return first(intersections);
```

Comment:
I know that is not part of this PR changes but I was wondering why using the lodash first() as usually just getting the first element from an array is way faster than using the lodash version. Unless I'm missing some context or some info here :D
Just for the sake of cleanness?

Reply:
It bounds checks under the hood, so it will be a little slower. But I have to bounds check anyway in the case where len is 0.